python/mrtrix3/commands/5ttgen/odf.py

```diff
@@ -0,0 +1,95 @@
+# Covered Software is free of defects, merchantable, fit for a
+# particular purpose or non-infringing.
+# See the Mozilla Public License v. 2.0 for more details.
+#
+# For more details, see http://www.mrtrix.org/.
+
+import os
+import subprocess
+from mrtrix3 import MRtrixError
+from mrtrix3 import app, path, run
+
+
+def usage(base_parser, subparsers):  # pylint: disable=unused-variable
+    parser = subparsers.add_parser('odf', parents=[base_parser])
+    parser.set_author("Arkiev D'Souza (arkiev.dsouza@sydney.edu.au) & Robert E. Smith (robert.smith@florey.edu.au)")
+    parser.set_synopsis('Generate a 5TT image from ODF images.')
+    
+    parser.add_argument('template_odf_wm', type=app.Parser.ImageIn(), help='The input white-matter ODF')
+    parser.add_argument('template_odf_gm', type=app.Parser.ImageIn(), help='The input grey-matter ODF')
+    parser.add_argument('template_odf_csf', type=app.Parser.ImageIn(), help='The input cerebrospinal fluid ODF')
+    parser.add_argument('mask_image', type=app.Parser.ImageIn(), help='The input binary brain mask image')
+    parser.add_argument('fTT_image', type=app.Parser.ImageOut(), help='The output 5TT image')
+
+    options = parser.add_argument_group('Options specific to the "odf" algorithm')
+    options.add_argument('-sgm', nargs='?', default=None, type=app.Parser.ImageIn(), help='Provide a mask of voxels that should be set to sub-cortical grey matter')
+    options.add_argument('-pathology', nargs='?', default=None, type=app.Parser.ImageIn(), help='Provide a mask of voxels that should be set to pathological tissue')
+
+
+def check_file(filepath):
+    if not os.path.isfile(filepath):
+        raise MRtrixError(f'Required input file missing (expected location: {filepath})')
+
+
+def execute():  # pylint: disable=unused-variable
+    # Check for input files
+    check_file(app.ARGS.template_odf_wm)
+    check_file(app.ARGS.template_odf_gm)
+    check_file(app.ARGS.template_odf_csf)
+    check_file(app.ARGS.mask_image)
+
+    # Extract l=0 term from WM
+    run.command(f'mrconvert', app.ARGS.template_odf_wm, '-coord 3 0 -axes 0,1,2 wm_vol.mif')
+
+    # Set negative values to zero
+    run.command(f'mrcalc wm_vol.mif 0.0 -max wm_vol_pos.mif')
+    run.command(f'mrcalc', app.ARGS.template_odf_gm, '0.0 -max gm_vol_pos.mif')
+    run.command(f'mrcalc', app.ARGS.template_odf_csf, '0.0 -max csf_vol_pos.mif')
+
+    # Normalize each volume
+    run.command(f'mrmath wm_vol_pos.mif gm_vol_pos.mif csf_vol_pos.mif sum totalvol.mif')
+    run.command(f'mrcalc wm_vol_pos.mif totalvol.mif -divide wm_vol_pos_norm.mif')
+    run.command(f'mrcalc gm_vol_pos.mif totalvol.mif -divide gm_vol_pos_norm.mif')
+    run.command(f'mrcalc csf_vol_pos.mif totalvol.mif -divide csf_vol_pos_norm.mif')
+
+    # Create empty volume for SGM and pathology
+    run.command(f'mrcalc wm_vol.mif 0 -neq wm_vol_1.mif')
+    run.command(f'mrcalc wm_vol_1.mif 1 0 -replace empty_vol.mif')
```

Comment:
This could instead be done as:

```suggestion
    run.command(f'mrcalc wm_vol.mif 0 -neq - | '
                f'mrcalc - 1 0 -replace empty_vol.mif')
````

(Since the intermediate image isn't used anywhere else)

Or better, something like:

```suggestion
    run.command(f'mrcalc wm_vol.mif inf -gt empty_vol.mif')
````